pkg/api/converterfromapi.go

```diff
@@ -1049,6 +1052,9 @@ func convertAgentPoolProfileToVLabs(api *AgentPoolProfile, p *vlabs.AgentPoolPro
 		p.ImageRef = &vlabs.ImageReference{}
 		p.ImageRef.Name = api.ImageRef.Name
```

Comment:
this isn't covered by unit tests

Reply:
I added an agent pool to cover this in to/from conversion test, vlabs version only.